syft/frameworks/torch/hook/hook.py

```diff
@@ -492,6 +493,150 @@ def _hook_crypten(self):
+
+        crypten.nn.Module.send = module_send_
+        crypten.nn.Module.send_ = module_send_
+
+        def module_move_(nn_self, destination):
+            nn_self._check_encrypted()
+            params = list(nn_self.parameters())
+            for p in params:
+                p.move(destination)
+
+        crypten.nn.Module.move = module_move_
+
+        def module_get_(nn_self):
+            """overloads torch.nn instances with get method so that parameters could be sent back to owner"""
+            nn_self._check_encrypted()
+            for p in nn_self.parameters():
+                p.get_()
+
+            if isinstance(nn_self.forward, Plan):
+                nn_self.forward.get()
+
+            return nn_self
+
+        crypten.nn.Module.get_ = module_get_
+        crypten.nn.Module.get = module_get_
+
+        def module_share_(nn_self, *args, **kwargs):
+            """Overloads fix_precision for torch.nn.Module."""
+            # TODO: add .data and .grad to syft tensors
+            nn_self._check_encrypted()
+            if module_is_missing_grad(nn_self):
+                create_grad_objects(nn_self)
+
+            for p in nn_self.parameters():
+                p.share_(*args, **kwargs)
+
+            return nn_self
+
+        crypten.nn.Module.share_ = module_share_
+        crypten.nn.Module.share = module_share_
+
+        def module_fix_precision_(nn_self, *args, **kwargs):
+            """Overloads fix_precision for torch.nn.Module."""
+            nn_self._check_encrypted()
+            if module_is_missing_grad(nn_self):
+                create_grad_objects(nn_self)
+
+            for p in nn_self.parameters():
+                p.fix_precision_(*args, **kwargs)
+
+            return nn_self
+
+        crypten.nn.Module.fix_precision_ = module_fix_precision_
+        crypten.nn.Module.fix_precision = module_fix_precision_
+        crypten.nn.Module.fix_prec = module_fix_precision_
+
+        def module_float_precision_(nn_self):
+            """Overloads float_precision for torch.nn.Module, convert fix_precision
+            parameters to normal float parameters"""
+            # TODO: add .data and .grad to syft tensors
+            # if module_is_missing_grad(nn_self):
+            #    create_grad_objects(nn_self)
+            nn_self._check_encrypted()
+            for p in nn_self.parameters():
+                p.float_precision_()
+
+            return nn_self
+
+        crypten.nn.Module.float_precision_ = module_float_precision_
+        crypten.nn.Module.float_precision = module_float_precision_
+        crypten.nn.Module.float_prec = module_float_precision_
+
+        def module_copy(nn_self):
+            """Returns a copy of a torch.nn.Module"""
+            nn_self._check_encrypted()
+            return copy.deepcopy(nn_self)
+
+        crypten.nn.Module.copy = module_copy
+
+        @property
+        def owner(nn_self):
+            nn_self._check_encrypted()
+            for p in nn_self.parameters():
```

Comment:
Q: Can't we simply do ```return nn_self.parameters()[0].owner```

Reply:
Pretty much the same, I just prefered followed the same style with the hooking of torch.nn